src/main/java/command/CommandParser.java

```diff
@@ -0,0 +1,131 @@
+package command;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+
+import exceptions.InvalidDateTimeFormat;
+import exceptions.InvalidIndexException;
+import exceptions.MissingDescriptionException;
+import exceptions.ThorndikeException;
+import task.Deadline;
+import task.Event;
+import task.Todo;
+import utility.DateTimeParser;
+import utility.StringUtility;
+
+/**
+ * Utility class responsible for parsing user input and generating corresponding
+ * Command objects.
+ *
+ * @author Ho Khee Wei
+ */
+public abstract class CommandParser {
+
+    /**
+     * Parses the user input to create a corresponding Command object.
+     *
+     * @param input User input string to be parsed.
+     * @return A Command object corresponding to the parsed input.
+     * @throws ThorndikeException If there are errors in parsing or the input is
+     *                            invalid.
+     */
+    public static Command parse(String input) throws ThorndikeException {
+        String command = input.split(" ")[0];
+        String[] argList = input.split(" /");
+        String description = StringUtility.removeFirstWord(input.split(" /")[0]);
+        Map<String, String> arguments = new HashMap<>();
+        for (String arg : argList) {
+            String option = StringUtility.getFirstWord(arg);
+            String value = StringUtility.removeFirstWord(arg);
+
+            if (!option.equals(command)) {
+                arguments.put("command", command);
+                arguments.put("description", value);
+                arguments.put(option, value);
+            } else {
+
+            }
+        }
+
+        if (command.equals("list")) {
```

Comment:
Could consider the use of enums? E.g.

private static final String COMMAND_LIST = "list";

if (command.equals(COMMAND_LIST)) {
    return new CmdList();

Reply:
I have considered it but decided not to because the amount of times I used it is quite few, usually 1 or 2 times for variable. Thanks for the suggestion anyway!